realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -797,6 +829,28 @@ final Builder schema(Class<? extends RealmModel> firstClass, Class<? extends Rea
             return this;
         }
 
+        /**
+         * Sets whether or not users are allowed to perform calls to {@link Realm#executeTransaction} from the UI thread.
+         * <p>
```

Comment:
The comment you had here is actually pretty great at describing the issue and work-around. I would add that description to this doc as well: https://github.com/realm/realm-java/pull/7123/files#diff-b77f99deee1ca52ca3dc3e47554e4e50R1531